packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -284,8 +284,26 @@ class RunCommand extends RunCommandBase {
     }
 
     for (Device device in devices) {
-      if (await device.isLocalEmulator && !isEmulatorBuildMode(getBuildMode()))
-        throwToolExit('${toTitleCase(getModeName(getBuildMode()))} mode is not supported for emulators.');
+      if (await device.isLocalEmulator) {
+        if (await device.supportsHardwareRendering) {
+          final bool enableSoftwareRendering = argResults['enable-software-rendering'] == true;
+          if (enableSoftwareRendering) {
+            printStatus(
+              'Using software rendering with device ${device.name}. You may get better performance'
```

Comment:
you're missing a space between "performance" and "with" in this string literal, so it'll render as "better performancewith hardware".